mockld/polling_service.go

```diff
@@ -0,0 +1,84 @@
+package mockld
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/launchdarkly/sdk-test-harness/framework"
+
+	"github.com/gorilla/mux"
+)
+
+type PollingService struct {
+	sdkKind     SDKKind
+	currentData SDKData
+	currentEtag string
```

Comment:
I haven't done any Etag stuff in tests yet but this component supports it.

Reply:
In fact, as I realized later, there's no way to do anything useful regarding Etags now due to SDK limitations: that is, the SDKs all enforce a minimum polling interval of 30 seconds, so we can't do any caching-related tests that would involve more than one poll unless we create some loophole for that.

Reply:
It's absolutely possible to build such a test, but because of how SDK configuration works— and the fact that the test service can only use the same APIs that the customer can use (it doesn't have access to any kind of private APIs)— such a test would take at least 30 seconds to run.

Reply:
And, that in itself would not be a terribly valuable test, because if there's a mistake in our implementation of polling, the symptom is very unlikely to be "the SDK stores the results from the first poll, but it doesn't store the results from the second poll." I mean, theoretically anything's possible, but the kinds of things we'd be more interested in testing would be things like:

1. If the first poll got a recoverable error, it correctly polls again 30 seconds later
2. 1. If the first poll got an unrecoverable error, initialization fails and it does not poll again 30 seconds later
3. If the first poll succeeds, and the second one gets a recoverable error, it does a 3rd poll 30 seconds later
4. If the first poll succeeds, and the second one gets an recoverable error, it does not do a 3rd poll 30 seconds later
5. If the second poll returns a new Etag, that Etag value is used for the 3rd request 30 seconds later

1 & 2 would take at least 30 seconds; 3, 4, & 5 would take at least _60_ seconds.